raven/processes/wps_nalcms_zonal_stats.py

```diff
@@ -0,0 +1,144 @@
+            status_supported=True,
+            store_supported=True)
+
+    def _handler(self, request, response):
+
+        shape_url = request.inputs['shape'][0].file
+        band = request.inputs['band'][0].data
+        geojson_out = request.inputs['return_geojson'][0].data
+        touches = request.inputs['select_all_touching'][0].data
+
+        vectors = ['.gml', '.shp', '.gpkg', '.geojson', '.json']
+        vector_file = single_file_check(archive_sniffer(shape_url, working_dir=self.workdir, extensions=vectors))
+        vec_crs = crs_sniffer(vector_file)
+
+        if 'raster' in request.inputs:  # For raster files using the UNFAO Land Cover Classification System (19 types)
+            rasters = ['.tiff', '.tif']
+            raster_url = request.inputs['raster'][0].file
+            raster_file = single_file_check(archive_sniffer(raster_url, working_dir=self.workdir, extensions=rasters))
+            ras_crs = crs_sniffer(raster_file)
+
+            if vec_crs != ras_crs:
+                msg = 'CRS for files {} and {} are not the same. Reprojecting...'.format(vector_file, raster_file)
+                LOGGER.warning(msg)
+
+                # Reproject full vector to preserve feature attributes
+                projected = tempfile.NamedTemporaryFile(prefix='reprojected_', suffix='.json', delete=False,
+                                                        dir=self.workdir).name
+                generic_vector_reproject(vector_file, projected, driver='GeoJSON', source_crs=vec_crs,
+                                         target_crs=ras_crs)
+                vector_file = projected
+
+        else:  # using the NALCMS data from GeoServer
+            projected = tempfile.NamedTemporaryFile(prefix='reprojected_', suffix='.json', delete=False,
+                                                    dir=self.workdir).name
+            generic_vector_reproject(vector_file, projected, driver='GeoJSON', source_crs=vec_crs,
+                                     target_crs=NALCMS_PROJ4)
+            vector_file = projected
+
+            bbox = gis.get_bbox(projected)
+            raster_url = 'public:EarthEnv_DEM90_NorthAmerica'
+            raster_bytes = gis.get_nalcms_wcs(bbox)
+            raster_file = tempfile.NamedTemporaryFile(prefix='wcs_', suffix='.tiff', delete=False,
+                                                      dir=self.workdir).name
+            with open(raster_file, 'wb') as f:
+                f.write(raster_bytes)
+
+        try:
+            stats = zonal_stats(
+                vector_file, raster_file, stats=['count', 'nodata'],
+                band=band, categorical=True, category_map=NALCMS_CATEGORIES, all_touched=touches,
+                geojson_out=geojson_out, raster_out=False)
+
+            if not geojson_out:
+                response.outputs['statistics'].data = json.dumps(stats)
+
+            else:
+                feature_collect = {'type': 'FeatureCollection', 'features': stats}
```

Comment:
Am I correct thinking the output's type depends on `geojson_out` ? As discussed earlier, that's not ideal.

Reply:
Ah yes, this was changed in the other branch. I'll update this here as well.

Reply:
Would it make sense to split the functions based on the desired output? If hydro models don't need geographic information, they can use `wps_simple_zonal_stat` and for functions that do, use `wps_geojson_zonal_stat`?

Reply:
Not clear to me what would be best. What are the typical use cases ? And also, I'd like to avoid code duplication, and to do that, we'd need a base class. So let's keep it simple for now, and think about improvements for later.

Reply:
I would tend towards preserving the GeoJSON format since that provides all the same information and preserves the geometry, but I'm biased. Also, saving this information within a `FeatureCollection` provides an easier means of loading and parsing information for each feature.

Reply:
Ok, I just don't want the output's type to change based on arguments. Let's return a FeatureCollection all the time and users will have to dig inside it to get the stats. Or split these two outputs entirely, whatever sounds most convenient.

Reply:
Last question before I push changes: Can we specify in the `wps_nalcms_zonal_stats.py` to output either the simplified values or the real values for pixels? I can see having the real categories being useful for other applications.

Reply:
Yes, makes sense.